card-games-vite/src/components/App/App.jsx

```diff
@@ -0,0 +1,394 @@
+  const [gameInfo, setGameInfo] = useState([]);
+  const [isDrawPileEmpty, setIsDrawPileEmpty] = useState(false);
+  const [isDiscardPileEmpty, setIsDiscardPileEmpty] = useState(true);
+
+  const handleEditProfileClick = () => {
+    setActiveModal("edit-profile-modal");
+  };
+  const handleLoginClick = () => {
+    setActiveModal("login-modal");
+  };
+  const handleRegistrationClick = () => {
+    setActiveModal("signup-modal");
+  };
+
+  const handleDiscardPileClick = () => {
+    setActiveModal("discard-modal");
+  };
+
+  const handleCloseModal = () => {
+    setActiveModal("");
+  };
+
+  const isEditProfileModalOpen = activeModal === "edit-profile-modal";
+  const isLoginModalOpen = activeModal === "login-modal";
+  const isRegistrationModalOpen = activeModal === "signup-modal";
+  const isDiscardModalOpen = activeModal === "discard-modal";
+
+  const handleLogin = ({ email, password }, resetForm) => {
+    if (!email || !password) {
+      return;
+    }
+
+    authorize(email, password)
+      .then((data) => {
+        if (data.token) {
+          localStorage.setItem("jwt", data.token);
+          setIsLoggedIn(true);
+          setUser(data);
+          handleCloseModal();
+          resetForm();
+        }
+      })
+      .catch(console.error);
+  };
+
+  const handleRegistration = ({ email, password, name, avatar }, resetForm) => {
+    register(email, password, name, avatar)
+      .then((data) => {
+        if (data.name) {
+          handleLogin({ email, password }, resetForm);
+          resetForm();
+          handleCloseModal();
+        }
+      })
+      .catch(console.error);
+  };
+
+  const handleLogout = () => {
+    setIsLoggedIn(false);
+    localStorage.removeItem("jwt");
+    setUserData({});
+  };
+
+  const handleEditProfile = ({ name, avatar }, resetForm) => {
+    editProfileInfo(name, avatar)
+      .then((profile) => {
+        setUserData({
+          name: profile.name,
+          avatar: profile.avatar,
+          email: profile.email,
+          id: profile.id,
+        });
+        setCurrentUser(profile);
+        resetForm();
+        handleCloseModal();
+      })
+      .catch((err) => console.error);
+  };
+
+  const setUser = (token) => {
+    checkToken(token)
+      .then((user) => {
+        setCurrentUser(user);
+        setIsLoggedIn(true);
+      })
+      .catch((err) => {
+        console.error(err);
+        setIsLoggedIn(false);
+      });
+  };
+
+  const handleCardLike = (game) => {
+    const setGameCardLikes = (updatedGame) => {
+      setGameInfo((games) => {
+        return games.map((item) =>
+          item.id === updatedGame.id ? updatedGame : item
+        );
+      });
+    };
+
+    setIsLoading(true);
+    if (game.liked) {
+      dislikeGame(game.id, localStorage.getItem("jwt"))
+        .then((res) => {
+          game.liked = res.liked;
+          setGameCardLikes(game);
+        })
+        .catch(console.error)
+        .finally(setIsLoading(false));
```

Comment:
`        .finally(setIsLoading(false));
`

In `then`, `catch`, `finally` you need to pass only the **reference** to the function (not a function call) otherwise the function is called immediately without waiting for the response from the server. You can pass an **unnamed** arrow function as well. 


```js

        .finally(() => setIsLoading(false));

```